### PR TITLE
Display full rate limit errors message from API response instead our constructing our own

### LINKED ISSUE #2738: Display rate limit errors message from API response
In order to make the rate limit exceptions more clear, we should extract and output the message returned by the API when a rate limit happens instead of outputting a generic error.

https://github.com/zeit/now-cli/blob/413fdd7c39949805ec8cc40d1308d9d6b6c459a7/src/util/handle-error.ts#L34-L38

### PR DESCRIPTION
**Pull request by TooTallNate**
Closes #2738.

### CODE REVIEW

File: src/util/handle-error.ts [hunk 1]
```diff
@@ -23,22 +23,9 @@ export default function handleError(
...
-        errorOutput(
-          `Rate limit exceeded error. Try again in ${ms(
-            ((<APIError>error).retryAfter as number) * 1000,
-            { long: true }
-          )}, or upgrade your plan here: ${chalk.cyan('https://zeit.co/account/plan')}`
-        )
-      );
-    }
+    // Rate limited: display the message from the server-side,
+    // which containsmore details
```
Comment (styfle) [inline src/util/handle-error.ts]:
```suggestion
    // which contains more details
```

File: src/util/handle-error.ts [hunk 2]
```diff
@@ -23,22 +23,9 @@ export default function handleError(
       errorOutput('Authentication error. Run `now login` to log-in again.')
     );
   } else if ((<APIError>error).status === 429) {
```
Comment (styfle) [inline src/util/handle-error.ts:25]: I realize you didn't change this code, but casting to `APIError` is a code smell.

Can you instead wrap this whole if statement with `if (typeof error.status === 'number')` which should narrow the type?

Comment (TooTallNate) [reply src/util/handle-error.ts:25]: That wouldn't narrow it down actually, user-defined type guards need to be used instead. I actually did spend close to an hour trying to remove the type casts from this file but ran into issues regarding the `NowError` and its subclasses (see `errors-ts.ts`). It was kind of a nightmare, and I ended up aborting mission.

IMO we need to re-think the generics of the `NowError` class and make it more… typeable, and DRY.

File: src/util/handle-error.ts (hunk 1 above)
Comment (TooTallNate) [reply src/util/handle-error.ts]: Thanks, nice catch

File: src/util/handle-error.ts (hunk 2 above)
Comment (TooTallNate) [reply src/util/handle-error.ts:25]: Also there's a lot more casting in this file than just this one. Territory for a separate PR I think.

**Review by styfle — APPROVED** [review]

**Pull request merged**